components/Tabs/tests/unit/MDCTabBarLayoutTests.m

```diff
@@ -0,0 +1,136 @@
+  MDCItemBar *itemBar = nil;
+  for (UIView *subview in tabBar.subviews) {
+    if ([subview isKindOfClass:[MDCItemBar class]]) {
+      itemBar = (MDCItemBar *)subview;
+    }
+  }
+  UICollectionView *collectionView = nil;
+  for (UIView *subview in itemBar.subviews) {
+    if ([subview isKindOfClass:[UICollectionView class]]) {
+      collectionView = (UICollectionView *)subview;
+    }
+  }
+  return collectionView;
+}
+
+// Returns the visible items of the given collectionView in sorted order.
+static NSArray<UICollectionViewCell *> *SortedCellsFromCollectionView(
+    UICollectionView *collectionView) {
+  NSArray<NSIndexPath *> *indexPathsForVisibleItems = collectionView.indexPathsForVisibleItems;
+  NSArray<NSIndexPath *> *sortedIndexPaths =
+      [indexPathsForVisibleItems sortedArrayUsingSelector:@selector(compare:)];
+  NSMutableArray *sortedCells = [NSMutableArray array];
+  for (NSIndexPath *indexPath in sortedIndexPaths) {
+    UICollectionViewCell *cell = [collectionView cellForItemAtIndexPath:indexPath];
+    if (cell) {
+      [sortedCells addObject:cell];
+    }
+  }
+  return sortedCells;
+}
+
+@interface MDCTabBarLayoutTests : XCTestCase
+
+@end
+
+@implementation MDCTabBarLayoutTests {
+  MDCTabBar *_tabBar;
+}
+
+- (void)setUp {
+  [super setUp];
+
+  // Create a tabbar with some dummy items and a dummy frame.
+  _tabBar = [[MDCTabBar alloc] initWithFrame:CGRectZero];
+  CGFloat tabBarHeight = [MDCTabBar defaultHeightForItemAppearance:_tabBar.itemAppearance];
+  _tabBar.frame = CGRectMake(0, 0, 200, tabBarHeight);
+  UITabBarItem *item1 = [[UITabBarItem alloc] initWithTitle:@"first" image:nil tag:0];
+  UITabBarItem *item2 = [[UITabBarItem alloc] initWithTitle:@"second" image:nil tag:0];
+  _tabBar.items = @[ item1, item2 ];
+  [_tabBar setNeedsLayout];
+  [_tabBar layoutIfNeeded];
+}
+
+- (void)tearDown {
+  // Explictly nil out ivars so that they are properly deallocated between tests. Otherwise XCTest
+  // will keep them alive until all test cases are complete.
+  _tabBar = nil;
+  [super tearDown];
+}
+
+// Tests the default layout behavior of of the tab bar's collection view.
+- (void)testLeftToRightLayout {
+  // Given
+  UICollectionView *collectionView = ExtractCollectionViewFromTabBar(_tabBar);
+  NSArray<UICollectionViewCell *> *sortedVisibleItems =
+      SortedCellsFromCollectionView(collectionView);
+  XCTAssertEqual(sortedVisibleItems.count, 2ul);
+  UICollectionViewCell *firstItemCell = sortedVisibleItems.firstObject;
+  UICollectionViewCell *secondItemCell = sortedVisibleItems.lastObject;
+  UICollectionViewFlowLayout *flowLayout =
+      (UICollectionViewFlowLayout *)collectionView.collectionViewLayout;
+
+  // When
+  _tabBar.mdf_semanticContentAttribute = UISemanticContentAttributeUnspecified;
+
+  // Then
+  CGFloat leftInset = flowLayout.sectionInset.left;
+  XCTAssertEqualWithAccuracy(firstItemCell.frame.origin.x, leftInset, (CGFloat)0.001);
```

Comment:
Please use `CGRectGetMinX()` instead of accessing the `origin` directly.  Same on the next line, and also use `CGRectGetWidth()` instead of accessing the `size`.

Reply:
Done.